src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -533,4 +533,88 @@ lemma norm_noninc_of_isometry (hf : isometry f) : f.norm_noninc :=
 
 end isometry
 
+variables {V₄ W₁ W₂ W₃ : Type*} [semi_normed_group V₄] [semi_normed_group W₁] [semi_normed_group W₂]
+  [semi_normed_group W₃]
+variables (f) (g : normed_group_hom V W)
+variables {f₁ g₁ : normed_group_hom V₁ W₁}
+variables {f₂ g₂ : normed_group_hom V₂ W₂}
+variables {f₃ g₃ : normed_group_hom V₃ W₃}
+
+/-- The equalizer of two morphisms `f g : normed_group_hom V W`. -/
+def equalizer := (f - g).ker
+
+namespace equalizer
+
+/-- The inclusion of `f.equalizer g` as a `normed_group_hom`. -/
+def ι : normed_group_hom (f.equalizer g) V := incl _
+
+lemma condition : f.comp (ι f g) = g.comp (ι f g) :=
+by { ext, rw [comp_apply, comp_apply, ← sub_eq_zero, ← normed_group_hom.sub_apply], exact x.2 }
+
+variables {f g}
+
+/-- If `φ : normed_group_hom V₁ V` is such that `f.comp φ = g.comp φ`, the induced morphism
+`normed_group_hom V₁ (f.equalizer g)`. -/
+@[simps]
+def lift (φ : normed_group_hom V₁ V) (h : f.comp φ = g.comp φ) :
+  normed_group_hom V₁ (f.equalizer g) :=
```

Comment:
Is this true in the reverse direction too? Is every `normed_group_hom V₁ (f.equalizer g)` uniquely associated with a `{φ : normed_group_hom V₁ V // f.comp φ = g.comp φ}`? If so, can you make this an equiv?

Reply:
I've added `lift_equiv`.